dirty_equals/_other.py

```diff
@@ -542,3 +543,149 @@ class Foo:
 
     def _post_init(self) -> None:
         self.strict = True
+
+
+class IsEnum(DirtyEquals[Enum]):
+    """
+    Checks if an instance is an Enum.
+    Inherits from [`DirtyEquals`][dirty_equals.DirtyEquals].
+    ```py title="IsEnum"
+    from enum import Enum, auto
+    from dirty_equals import IsEnum
+    class ExampleEnum(Enum):
+        a = auto()
+        b = auto()
+    a = ExampleEnum.a
+    assert a == IsEnum
+    assert a == IsEnum(ExampleEnum)
+    ```
+    """
+
+    def __init__(self, enum_cls: type[Enum] = Enum):
+        """
+        Args:
+            enum_cls: Enum class to check against.
+        """
+        self._enum_cls = enum_cls
+
+    def equals(self, other: Any) -> bool:
+        return isinstance(other, self._enum_cls)
+
```

Comment:
One idea to account for checking if `other` is a value in the `_enum_cls` could be:

```python
if isinstance(other, Enum):
    return isinstance(other, self._enum_cls)
else:
    return other in {i.value for i in self._enum_cls}
```

Leading to:
```py
from enum import Enum, auto

class ExampleEnum(Enum):
    a = auto()
    b = auto()

assert 1 == IsEnum(ExampleEnum)
assert 3 != IsEnum(ExampleEnum)
```

Reply:
yes we should check that, the simplest and most correct approach I think is to do

```py
try:
    self._enum_cls(other)
except ValueError:
    return False
else:
     return True
```

Reply:
Maybe even:
```py
try:
    _ = self._enum_cls(other)
    return True
except ValueError:
    return False
```
does the trick?

Reply:
Trying these 2 implementations, however it would lead to something like:
```py
ExampleEnum.a != IsEnum
```
which I don't think it intended behaviour.

I would stuck with the suggestion:
```py
if isinstance(other, Enum):
    return isinstance(other, self._enum_cls)
else:
    return other in {i.value for i in self._enum_cls}
```
and possibly "cache" the values `{i.value for i in self._enum_cls}` in the instantiation step.
Let me know what your thoughts are